cypress/support/commands.js

```
@@ -1,70 +1,49 @@
-// ***********************************************
-// This example commands.js shows you how to
-// create various custom commands and overwrite
-// existing commands.
-//
-// For more comprehensive examples of custom
-// commands please read more here:
-// https://on.cypress.io/custom-commands
-// ***********************************************
-//
-//
-// -- This is a parent command --
-// Cypress.Commands.add('login', (email, password) => { ... })
-//
-//
-// -- This is a child command --
-// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
-//
-//
-// -- This is a dual command --
-// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
-//
-//
-// -- This will overwrite an existing command --
-// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
-
 const imgUrl = 'https://static.productionready.io/images/smiley-cyrus.jpg';
 
-Cypress.Commands.add('login', (email, username, password) => {
-  cy.request('POST', '/api/users', {
-    user: {
-      email,
-      username,
-      password
-    }
-  }).then((response) => {
-    const user = {
-      bio: response.body.user.bio,
-      effectiveImage: imgUrl,
-      email: response.body.user.email,
-      image: response.body.user.image,
-      token: response.body.user.token,
-      username: response.body.user.username
-    };
-    window.localStorage.setItem('user', JSON.stringify(user));
-    cy.setCookie('auth', response.body.user.token);
+// Register a user (only once)
+Cypress.Commands.add('register', (email, username, password) => {
+  return cy.request({
+    method: 'POST',
+    url: '/api/users',
+    body: { user: { email, username, password } },
+    failOnStatusCode: false, // optional: ignore if user exists
```

Using `failOnStatusCode: false` will suppress all non-2xx responses for the register request. That can hide genuine problems (network, server errors) and make test failures harder to diagnose. If the intent is only to ignore "user already exists" responses, consider checking the response status and only swallowing the expected status (e.g. 409), or failing on other statuses so the test setup doesn't silently proceed with an unexpected state.